### PR TITLE
DataTable onChange replaced with onRowChange

### LINKED ISSUE #1407: Consider new callback for DataTable row selection
Related to #273.

* DataTable's `onChange` is `(selectedArrayOfItems, event)`. [ Event callbacks should pass in the synthetic event, then a data object with named key/values that relate to the event.](https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#event-callbacks) according the Codebase Overview.
* DataTable shouldn't be moved. Where would be move it to?
* DataTable is one of the most popular components. I don't want to break the row selection callback `onChange`.
* Possible solutions included a new callback `onRowChange`, `onRowSelect`, or `onSelect` with `onChange` being deprecated.

### PR DESCRIPTION
**Pull request by interactivellama**
Update parameters.

Fixes #1407

### Additional description

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/data-table/__examples__/advanced.jsx [hunk 1]
```diff
@@ -77,8 +77,9 @@ const Example = createReactClass({
 		};
 	},
 
-	handleChanged (selection) {
-		this.setState({ selection });
+	handleChanged (event, data) {
+		this.setState({ selection: data.selection });
+		console.log(event, data);
```
Comment (futuremint) [inline components/data-table/__examples__/advanced.jsx]: Are we allowing these in this codebase? (I always forget which project has which practices 😃)

File: components/data-table/index.jsx [hunk 2]
```diff
@@ -131,23 +131,29 @@ const DataTable = createReactClass({
...
 		 * A variant which removes hover style on rows
 		 */
 		noRowHover: PropTypes.bool,
 		/**
-		 * This function fires when the selection of rows changes.
+		 * This function fires when the selection of rows changes. This component passes in `event, { selection }` to the function. `selection` is an array of objects from the `items` prop.
+		 *
+		 * This used to be `onChange` which is deprecated now, so that the parameters can be consistent with other components. `onChange` passed in the selection first and the event wtihout a data object.
 		 */
-		onChange: PropTypes.func,
```
Comment (futuremint) [inline components/data-table/index.jsx]: Is this _removing_ the onChange prop?  Shouldn't we leave it in if its only deprecated, and then remove it later?

File: components/data-table/__examples__/advanced.jsx (hunk 1 above)
Comment (futuremint) [reply components/data-table/__examples__/advanced.jsx]: nm, makes sense here in the example if one wanted to see how stuff is working.

Comment (interactivellama) [reply components/data-table/__examples__/advanced.jsx]: Yep, here is where they are allowed to pass lint. https://github.com/salesforce/design-system-react/blob/master/.eslintrc.json#L22

**Pull request merged**